docs/etcd-quorum-guard.md

```diff
@@ -0,0 +1,39 @@
+# etcd Quorum Guard
+
+The etcd Quorum Guard ensures that quorum is maintained for etcd for
+[OpenShift](https://openshift.io/).
+
+For the etcd cluster to remain usable, we must maintain quorum, which
+is a majority of all etcd members.  For example, an etcd cluster with
+3 members (i.e. a 3 master deployment) must have at least 2 healthy
+etcd member to meet the quorum limit.
+
+There are situations where 2 etcd members could be down at once:
+
+* a master has gone offline and the MachineConfig Controller (MCC)
+  tries to rollout a new MachineConfig (MC) by rebooting masters
+* the MCC is doing a MachineConfig rollout and doesn't wait for the
+  etcd on the previous master to become healthy again before rebooting
+  the next master
+
+In short, we need a way to ensure that a drain on a master is not
```

Comment:
"The etcd Quorum Guard ensures that a drain on a master is not allowed to proceed if the reboot of the master would cause etcd quorum loss.  It is implemented as a deployment, with one pod per master node."

Reply:
OK